xtask/src/main.rs

```diff
@@ -1,35 +1,232 @@
+    repetition_count: u32,
+    /// Whether to run the benchmarks in debug mode.
+    #[arg(long, short = 'g')]
+    debug: bool,
+}
 
-Tasks:
-    bump-version
-"
-    );
+#[derive(Args)]
+struct BuildWasm {
+    /// Compile the library more quickly, with fewer optimizations
+    /// and more runtime assertions.
+    #[arg(long, short = '0')]
+    debug: bool,
+    /// Run emscripten using docker, even if \`emcc\` is installed.
+    /// By default, \`emcc\` will be run directly when available.
+    #[arg(long, short)]
+    docker: bool,
+    /// Run emscripten with verbose output.
+    #[arg(long, short)]
+    verbose: bool,
 }
 
-fn main() -> Result<(), Box<dyn std::error::Error>> {
-    let Some(task) = std::env::args().nth(1) else {
-        print_help();
-        std::process::exit(0);
-    };
+#[derive(Args)]
+struct BumpVersion {
+    /// The version to bump to.
+    #[arg(long, short)]
+    version: Option<Version>,
+}
 
-    match task.as_str() {
-        "bump-version" => {
-            bump_versions()?;
+#[derive(Args)]
+struct Clippy {
+    /// Automatically apply lint suggestions (`clippy --fix`).
+    #[arg(long, short)]
+    fix: bool,
+    /// The package to run Clippy against (`cargo -p <PACKAGE> clippy`).
+    #[arg(long, short)]
+    package: Option<String>,
+}
+
+#[derive(Args)]
+struct GenerateFixtures {
+    /// Generates the parser to WASM
+    #[arg(long, short)]
+    wasm: bool,
+    /// Run emscripten via docker even if it is installed locally.
+    #[arg(long, short, requires = "wasm")]
+    docker: bool,
+}
+
+#[derive(Args)]
+struct Test {
+    /// Compile C code with the Clang address sanitizer.
+    #[arg(long, short)]
+    address_sanitizer: bool,
+    /// Run only the corpus tests whose name contain the given string.
+    #[arg(long, short)]
+    example: Option<String>,
+    /// Run the given number of iterations of randomized tests (default 10).
+    #[arg(long, short)]
+    iterations: Option<u32>,
+    /// Set the seed used to control random behavior.
+    #[arg(long, short)]
+    seed: Option<u32>,
+    /// Print parsing log to stderr.
+    #[arg(long, short)]
+    debug: bool,
+    /// Generate an SVG graph of parsing logs.
+    #[arg(long, short = 'D')]
+    debug_graph: bool,
+    /// Run the tests with a debugger.
+    #[arg(short)]
+    g: bool,
+    #[arg(trailing_var_arg = true)]
+    args: Vec<String>,
+}
+
+const BUILD_VERSION: &str = env!("CARGO_PKG_VERSION");
+const BUILD_SHA: Option<&str> = option_env!("BUILD_SHA");
+const EMSCRIPTEN_VERSION: &str = include_str!("../../cli/loader/emscripten-version");
+const EMSCRIPTEN_TAG: &str = concat!(
+    "docker.io/emscripten/emsdk:",
+    include_str!("../../cli/loader/emscripten-version")
+);
+
+fn main() {
+    let result = run();
+    if let Err(err) = &result {
+        // Ignore BrokenPipe errors
+        if let Some(error) = err.downcast_ref::<std::io::Error>() {
+            if error.kind() == std::io::ErrorKind::BrokenPipe {
```

Comment:
Why?

Reply:
copied from [cli/src/main.rs](https://github.com/tree-sitter/tree-sitter/blob/5c6445edea1efaa2cac45b2fd16c42441896a11a/cli/src/main.rs#L1272), because of https://github.com/tree-sitter/tree-sitter/pull/1135. Though I think xtask commands are less likely to be piped, I don't see why not to have this